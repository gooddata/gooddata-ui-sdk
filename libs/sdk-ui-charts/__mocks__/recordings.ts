// (C) 2020 GoodData Corporation

import { DataViewFacade } from "@gooddata/sdk-ui";
import { ScenarioRecording, recordedDataView } from "@gooddata/sdk-backend-mockingbird";

export function recordedDataFacade(rec: ScenarioRecording): DataViewFacade {
    return DataViewFacade.for(recordedDataView(rec));
}
