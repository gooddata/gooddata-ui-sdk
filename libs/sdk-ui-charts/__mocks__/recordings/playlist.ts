// (C) 2019-2019 GoodData Corporation

import { createRequire } from "module";
const require = createRequire(import.meta.url);

// THIS FILE WAS AUTO-GENERATED ON 2019-11-05T10:50:00.366Z; DO NOT EDIT; LOOK INTO 'mock-handling' TOOLING

export const AreaChartWith3MetricsAndViewByAttribute = {
    definition: require("./execution/area_chart_with_3_metrics_and_view_by_attribute/definition.json"),
    response: require("./execution/area_chart_with_3_metrics_and_view_by_attribute/response.json"),
    result: require("./execution/area_chart_with_3_metrics_and_view_by_attribute/result.json"),
};

export const AreaChartWithMeasureViewByAndStackBy = {
    definition: require("./execution/area_chart_with_measure_view_by_and_stack_by/definition.json"),
    response: require("./execution/area_chart_with_measure_view_by_and_stack_by/response.json"),
    result: require("./execution/area_chart_with_measure_view_by_and_stack_by/result.json"),
};

export const BarChartWith3MetricsAndViewByAttribute = {
    definition: require("./execution/bar_chart_with_3_metrics_and_view_by_attribute/definition.json"),
    response: require("./execution/bar_chart_with_3_metrics_and_view_by_attribute/response.json"),
    result: require("./execution/bar_chart_with_3_metrics_and_view_by_attribute/result.json"),
};

export const BarChartWith3MetricsAndViewByAttributeFunformat = {
    definition: require("./execution/bar_chart_with_3_metrics_and_view_by_attribute_funformat/definition.json"),
    response: require("./execution/bar_chart_with_3_metrics_and_view_by_attribute_funformat/response.json"),
    result: require("./execution/bar_chart_with_3_metrics_and_view_by_attribute_funformat/result.json"),
};

export const BarChartWith3MetricsAndViewByAttributePercInFormat = {
    definition: require("./execution/bar_chart_with_3_metrics_and_view_by_attribute_perc_in_format/definition.json"),
    response: require("./execution/bar_chart_with_3_metrics_and_view_by_attribute_perc_in_format/response.json"),
    result: require("./execution/bar_chart_with_3_metrics_and_view_by_attribute_perc_in_format/result.json"),
};

export const BarChartWith4MetricsAndViewByTwoAttributes = {
    definition: require("./execution/bar_chart_with_4_metrics_and_view_by_two_attributes/definition.json"),
    response: require("./execution/bar_chart_with_4_metrics_and_view_by_two_attributes/response.json"),
    result: require("./execution/bar_chart_with_4_metrics_and_view_by_two_attributes/result.json"),
};

export const BarChartWithPopMeasureAndViewByAttribute = {
    definition: require("./execution/bar_chart_with_pop_measure_and_view_by_attribute/definition.json"),
    response: require("./execution/bar_chart_with_pop_measure_and_view_by_attribute/response.json"),
    result: require("./execution/bar_chart_with_pop_measure_and_view_by_attribute/result.json"),
};

export const BarChartWithPopMeasureAndViewByAttributeX6 = {
    definition: require("./execution/bar_chart_with_pop_measure_and_view_by_attribute_x6/definition.json"),
    response: require("./execution/bar_chart_with_pop_measure_and_view_by_attribute_x6/response.json"),
    result: require("./execution/bar_chart_with_pop_measure_and_view_by_attribute_x6/result.json"),
};

export const BarChartWithPreviousPeriodMeasure = {
    definition: require("./execution/bar_chart_with_previous_period_measure/definition.json"),
    response: require("./execution/bar_chart_with_previous_period_measure/response.json"),
    result: require("./execution/bar_chart_with_previous_period_measure/result.json"),
};

export const BarChartWithPreviousPeriodMeasureX6 = {
    definition: require("./execution/bar_chart_with_previous_period_measure_x6/definition.json"),
    response: require("./execution/bar_chart_with_previous_period_measure_x6/response.json"),
    result: require("./execution/bar_chart_with_previous_period_measure_x6/result.json"),
};

export const BarChartWithSingleMeasureAndNoAttributes = {
    definition: require("./execution/bar_chart_with_single_measure_and_no_attributes/definition.json"),
    response: require("./execution/bar_chart_with_single_measure_and_no_attributes/response.json"),
    result: require("./execution/bar_chart_with_single_measure_and_no_attributes/result.json"),
};

export const BarChartWithStackByAndViewByAttributes = {
    definition: require("./execution/bar_chart_with_stack_by_and_view_by_attributes/definition.json"),
    response: require("./execution/bar_chart_with_stack_by_and_view_by_attributes/response.json"),
    result: require("./execution/bar_chart_with_stack_by_and_view_by_attributes/result.json"),
};

export const BarChartWithViewByAttribute = {
    definition: require("./execution/bar_chart_with_view_by_attribute/definition.json"),
    response: require("./execution/bar_chart_with_view_by_attribute/response.json"),
    result: require("./execution/bar_chart_with_view_by_attribute/result.json"),
};

export const BubbleChartWith1Metric = {
    definition: require("./execution/bubble_chart_with_1_metric/definition.json"),
    response: require("./execution/bubble_chart_with_1_metric/response.json"),
    result: require("./execution/bubble_chart_with_1_metric/result.json"),
};

export const BubbleChartWith2MetricsAndAttributeNoPrimaries = {
    definition: require("./execution/bubble_chart_with_2_metrics_and_attribute_no_primaries/definition.json"),
    response: require("./execution/bubble_chart_with_2_metrics_and_attribute_no_primaries/response.json"),
    result: require("./execution/bubble_chart_with_2_metrics_and_attribute_no_primaries/result.json"),
};

export const BubbleChartWith3MetricsAndAttribute = {
    definition: require("./execution/bubble_chart_with_3_metrics_and_attribute/definition.json"),
    response: require("./execution/bubble_chart_with_3_metrics_and_attribute/response.json"),
    result: require("./execution/bubble_chart_with_3_metrics_and_attribute/result.json"),
};

export const BubbleChartWith3MetricsAndAttributeNullsInData = {
    definition: require("./execution/bubble_chart_with_3_metrics_and_attribute_nulls_in_data/definition.json"),
    response: require("./execution/bubble_chart_with_3_metrics_and_attribute_nulls_in_data/response.json"),
    result: require("./execution/bubble_chart_with_3_metrics_and_attribute_nulls_in_data/result.json"),
};

export const ComboChartWithTwoMeasuresViewByAttribute = {
    definition: require("./execution/combo_chart_with_two_measures_view_by_attribute/definition.json"),
    response: require("./execution/combo_chart_with_two_measures_view_by_attribute/response.json"),
    result: require("./execution/combo_chart_with_two_measures_view_by_attribute/result.json"),
};

export const ComboChartWithTwoMeasuresViewByAttributeNoBuckets = {
    definition: require("./execution/combo_chart_with_two_measures_view_by_attribute_no_buckets/definition.json"),
    response: require("./execution/combo_chart_with_two_measures_view_by_attribute_no_buckets/response.json"),
    result: require("./execution/combo_chart_with_two_measures_view_by_attribute_no_buckets/result.json"),
};

export const ComboChartWithTwoMeasuresViewByAttributePercformat = {
    definition: require("./execution/combo_chart_with_two_measures_view_by_attribute_percformat/definition.json"),
    response: require("./execution/combo_chart_with_two_measures_view_by_attribute_percformat/response.json"),
    result: require("./execution/combo_chart_with_two_measures_view_by_attribute_percformat/result.json"),
};

export const HeadlineWithOneMeasure = {
    definition: require("./execution/headline_with_one_measure/definition.json"),
    response: require("./execution/headline_with_one_measure/response.json"),
    result: require("./execution/headline_with_one_measure/result.json"),
};

export const HeadlineWithOneMeasureWithIdentifier = {
    definition: require("./execution/headline_with_one_measure_with_identifier/definition.json"),
    response: require("./execution/headline_with_one_measure_with_identifier/response.json"),
    result: require("./execution/headline_with_one_measure_with_identifier/result.json"),
};

export const HeadlineWithTwoMeasures = {
    definition: require("./execution/headline_with_two_measures/definition.json"),
    response: require("./execution/headline_with_two_measures/response.json"),
    result: require("./execution/headline_with_two_measures/result.json"),
};

export const HeadlineWithTwoMeasuresBothEmpty = {
    definition: require("./execution/headline_with_two_measures_both_empty/definition.json"),
    response: require("./execution/headline_with_two_measures_both_empty/response.json"),
    result: require("./execution/headline_with_two_measures_both_empty/result.json"),
};

export const HeadlineWithTwoMeasuresBothSame = {
    definition: require("./execution/headline_with_two_measures_both_same/definition.json"),
    response: require("./execution/headline_with_two_measures_both_same/response.json"),
    result: require("./execution/headline_with_two_measures_both_same/result.json"),
};

export const HeadlineWithTwoMeasuresBothZero = {
    definition: require("./execution/headline_with_two_measures_both_zero/definition.json"),
    response: require("./execution/headline_with_two_measures_both_zero/response.json"),
    result: require("./execution/headline_with_two_measures_both_zero/result.json"),
};

export const HeadlineWithTwoMeasuresFirstEmpty = {
    definition: require("./execution/headline_with_two_measures_first_empty/definition.json"),
    response: require("./execution/headline_with_two_measures_first_empty/response.json"),
    result: require("./execution/headline_with_two_measures_first_empty/result.json"),
};

export const HeadlineWithTwoMeasuresFirstZero = {
    definition: require("./execution/headline_with_two_measures_first_zero/definition.json"),
    response: require("./execution/headline_with_two_measures_first_zero/response.json"),
    result: require("./execution/headline_with_two_measures_first_zero/result.json"),
};

export const HeadlineWithTwoMeasuresSecondEmpty = {
    definition: require("./execution/headline_with_two_measures_second_empty/definition.json"),
    response: require("./execution/headline_with_two_measures_second_empty/response.json"),
    result: require("./execution/headline_with_two_measures_second_empty/result.json"),
};

export const HeadlineWithTwoMeasuresSecondZero = {
    definition: require("./execution/headline_with_two_measures_second_zero/definition.json"),
    response: require("./execution/headline_with_two_measures_second_zero/response.json"),
    result: require("./execution/headline_with_two_measures_second_zero/result.json"),
};

export const HeadlineWithTwoMeasuresWithIdentifier = {
    definition: require("./execution/headline_with_two_measures_with_identifier/definition.json"),
    response: require("./execution/headline_with_two_measures_with_identifier/response.json"),
    result: require("./execution/headline_with_two_measures_with_identifier/result.json"),
};

export const HeatMapWithEmptyCells = {
    definition: require("./execution/heat_map_with_empty_cells/definition.json"),
    response: require("./execution/heat_map_with_empty_cells/response.json"),
    result: require("./execution/heat_map_with_empty_cells/result.json"),
};

export const HeatMapWithMetricRowColumn = {
    definition: require("./execution/heat_map_with_metric_row_column/definition.json"),
    response: require("./execution/heat_map_with_metric_row_column/response.json"),
    result: require("./execution/heat_map_with_metric_row_column/result.json"),
};

export const PieChartWithMetricsOnly = {
    definition: require("./execution/pie_chart_with_metrics_only/definition.json"),
    response: require("./execution/pie_chart_with_metrics_only/response.json"),
    result: require("./execution/pie_chart_with_metrics_only/result.json"),
};

export const PieChartWithMetricsOnlyFundata = {
    definition: require("./execution/pie_chart_with_metrics_only_fundata/definition.json"),
    response: require("./execution/pie_chart_with_metrics_only_fundata/response.json"),
    result: require("./execution/pie_chart_with_metrics_only_fundata/result.json"),
};

export const ScatterPlotWith2MetricsAndAttributeNullsInData = {
    definition: require("./execution/scatter_plot_with_2_metrics_and_attribute_nulls_in_data/definition.json"),
    response: require("./execution/scatter_plot_with_2_metrics_and_attribute_nulls_in_data/response.json"),
    result: require("./execution/scatter_plot_with_2_metrics_and_attribute_nulls_in_data/result.json"),
};

export const ScatterPlotWith2MetricsAndAttributeWithPrimary = {
    definition: require("./execution/scatter_plot_with_2_metrics_and_attribute_with_primary/definition.json"),
    response: require("./execution/scatter_plot_with_2_metrics_and_attribute_with_primary/response.json"),
    result: require("./execution/scatter_plot_with_2_metrics_and_attribute_with_primary/result.json"),
};

export const TreemapWithMetricAndStackByAttribute = {
    definition: require("./execution/treemap_with_metric_and_stack_by_attribute/definition.json"),
    response: require("./execution/treemap_with_metric_and_stack_by_attribute/response.json"),
    result: require("./execution/treemap_with_metric_and_stack_by_attribute/result.json"),
};

export const TreemapWithMetricAndViewByAttribute = {
    definition: require("./execution/treemap_with_metric_and_view_by_attribute/definition.json"),
    response: require("./execution/treemap_with_metric_and_view_by_attribute/response.json"),
    result: require("./execution/treemap_with_metric_and_view_by_attribute/result.json"),
};

export const TreemapWithMetricViewByAndStackByAttribute = {
    definition: require("./execution/treemap_with_metric_view_by_and_stack_by_attribute/definition.json"),
    response: require("./execution/treemap_with_metric_view_by_and_stack_by_attribute/response.json"),
    result: require("./execution/treemap_with_metric_view_by_and_stack_by_attribute/result.json"),
};

export const TreemapWithTwoMetricsAndStackByAttribute = {
    definition: require("./execution/treemap_with_two_metrics_and_stack_by_attribute/definition.json"),
    response: require("./execution/treemap_with_two_metrics_and_stack_by_attribute/response.json"),
    result: require("./execution/treemap_with_two_metrics_and_stack_by_attribute/result.json"),
};

// initialize recorded backend with this
export const MasterIndex = {
    testWorkspace: {
        execution: {
            fp_0b599ac8ef275e712d231f304d9e29d0: AreaChartWithMeasureViewByAndStackBy,
            fp_45eabf83c34eaa23f347b56a5858d57f: ComboChartWithTwoMeasuresViewByAttributeNoBuckets,
            fp_baa24e54ab2909e7a7ae4f5f6d21ffaa: BarChartWith3MetricsAndViewByAttributeFunformat,
            fp_d04ba7363a1ce9162b18fe0ddc005a07: BarChartWith3MetricsAndViewByAttributePercInFormat,
            fp_3f95917db9f15eb872d904631fc3edcd: BarChartWith4MetricsAndViewByTwoAttributes,
            fp_f567a925d82dd41eff9780eb4ca934d2: BarChartWithPopMeasureAndViewByAttribute,
            fp_62bcd4f8af06c91324c3af201560ac9c: BarChartWithPopMeasureAndViewByAttributeX6,
            fp_ac28e4e88b5011dd98c8c92f970f3910: BarChartWithPreviousPeriodMeasure,
            fp_b836c933d95c1aef2d07f950f38b645d: BarChartWithPreviousPeriodMeasureX6,
            fp_7ebc96ae6d02197155fcd63a41f870b3: BarChartWithSingleMeasureAndNoAttributes,
            fp_1b740fffd1df15049b4c5771cf293327: BarChartWithStackByAndViewByAttributes,
            fp_8d33743b38a06a7af0a1a71e3c573f8b: BarChartWithViewByAttribute,
            fp_4b146f4139daf56999dc16e76d478de9: BubbleChartWith1Metric,
            fp_14f9897772326f7e811f4aa9a20d871d: BubbleChartWith2MetricsAndAttributeNoPrimaries,
            fp_bcc9d43c8427eeb41548d866b1630374: BubbleChartWith3MetricsAndAttributeNullsInData,
            fp_63a093ee40d9e5d13e14cd186ced8376: ComboChartWithTwoMeasuresViewByAttributePercformat,
            fp_e2b71aa2be7ba9d7c46ee9f153cce9f2: HeadlineWithOneMeasure,
            fp_4f043c79dbefafee7e26f3f77d945b86: HeadlineWithOneMeasureWithIdentifier,
            fp_d53d09f0ed2ab5f3ef73b73ca749e4f7: HeadlineWithTwoMeasuresSecondZero,
            fp_4bf5193c9d1f6560bc5c43b0b2b99ea1: HeadlineWithTwoMeasuresWithIdentifier,
            fp_7ba00bd46fb37f68fbb549e997f349dd: HeatMapWithEmptyCells,
            fp_16dd93e14c96b0317d74f5e0e37792ff: HeatMapWithMetricRowColumn,
            fp_551204e9d10f29532c64d0aad6d9b19b: PieChartWithMetricsOnlyFundata,
            fp_f28b82a9ed53ff7655a633dca593e1fb: ScatterPlotWith2MetricsAndAttributeWithPrimary,
            fp_539276f8275d76bce58f585086fd2cf4: TreemapWithMetricViewByAndStackByAttribute,
            fp_d81a1d01654699ff102ae5be3839bcde: TreemapWithTwoMetricsAndStackByAttribute,
        },
    },
};
