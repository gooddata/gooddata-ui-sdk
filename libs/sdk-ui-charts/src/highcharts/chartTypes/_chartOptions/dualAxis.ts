// (C) 2019-2020 GoodData Corporation
import { measureLocalId } from "@gooddata/sdk-model";
import { IChartConfig } from "../../../interfaces/index.js";
import { BucketNames, DataViewFacade } from "@gooddata/sdk-ui";
import { isComboChart } from "../_util/common.js";

export function setMeasuresToSecondaryAxis(config: IChartConfig = {}, dv: DataViewFacade): IChartConfig {
    const isDualAxis = config.dualAxis ?? true;
    const type = config.type;
    const { secondary_yaxis: secondaryYAxis, ...remainConfig } = config;
    const secondaryMeasuresIds = dv.def().bucketMeasures(BucketNames.SECONDARY_MEASURES).map(measureLocalId);

    if (!isComboChart(type)) {
        return config;
    }

    if (!isDualAxis) {
        return remainConfig;
    }

    return {
        ...remainConfig,
        secondary_yaxis: {
            ...secondaryYAxis,
            measures: secondaryMeasuresIds,
        },
    };
}
