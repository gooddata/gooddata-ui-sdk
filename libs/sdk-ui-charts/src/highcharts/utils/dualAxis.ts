// (C) 2019 GoodData Corporation
import { DataViewFacade } from "@gooddata/sdk-backend-spi";
import { measureLocalId } from "@gooddata/sdk-model";
import { IChartConfig } from "../Config";
import { BucketNames, VisType } from "@gooddata/sdk-ui";
import { isComboChart } from "./common";
import get = require("lodash/get");

export function setMeasuresToSecondaryAxis(config: IChartConfig = {}, dv: DataViewFacade): IChartConfig {
    const isDualAxis: boolean = get(config, "dualAxis", true);
    const type: VisType = get(config, "type");
    const { secondary_yaxis: secondaryYAxis, ...remainConfig } = config;
    const secondaryMeasuresIds = dv.bucketMeasures(BucketNames.SECONDARY_MEASURES).map(measureLocalId);

    if (!isComboChart(type)) {
        return config;
    }

    if (!isDualAxis) {
        return remainConfig;
    }

    return {
        ...remainConfig,
        secondary_yaxis: {
            ...secondaryYAxis,
            measures: secondaryMeasuresIds,
        },
    };
}
