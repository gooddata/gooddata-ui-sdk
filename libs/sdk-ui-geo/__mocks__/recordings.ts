// (C) 2020 GoodData Corporation

import { DataViewFacade } from "@gooddata/sdk-ui";
import { ScenarioRecording, recordedDataView } from "@gooddata/sdk-backend-mockingbird";
import { ExamplesRecordings } from "@gooddata/live-examples-workspace";
import { getGeoData } from "../src/core/geoChart/helpers/geoChart/data";
import cloneDeep from "lodash/cloneDeep";
export function recordedDataFacade(rec: ScenarioRecording): DataViewFacade {
    return DataViewFacade.for(recordedDataView(rec));
}

function createShortcut(rec: ScenarioRecording) {
    const dv = recordedDataFacade(rec);

    return {
        dv,
        geoData: getGeoData(dv, "empty value", "null value"),
    };
}

function trimmedScenario(rec: ScenarioRecording, to: number[]) {
    const copy = cloneDeep(recordedDataView(rec));

    to.forEach((trim, dim) => {
        if (trim < 0) {
            return;
        }

        const headers = copy.headerItems[dim];

        if (headers) {
            for (let i = 0; i < headers.length; i++) {
                headers[i] = headers[i].slice(0, trim);
            }
        }

        if (dim === 0) {
            // @ts-ignore
            copy.data = copy.data.slice(0, trim) as any;
        } else if (dim === 1) {
            for (let i = 0; i < copy.data.length; i++) {
                const row = copy.data[i];

                if (Array.isArray(row)) {
                    copy.data[i] = row.slice(0, trim);
                }
            }
        }

        if (copy.count[dim]) {
            copy.count[dim] = trim;
        }

        if (copy.totalCount[dim]) {
            copy.totalCount[dim] = trim;
        }
    });

    const dv = DataViewFacade.for(copy);

    return {
        dv,
        geoData: getGeoData(dv, "empty value", "null value"),
    };
}

export const RecShortcuts = {
    LocationOnly: createShortcut(ExamplesRecordings.Scenarios.GeoPushpinChart.LocationOnly),
    LocationOnlySmall: trimmedScenario(ExamplesRecordings.Scenarios.GeoPushpinChart.LocationOnly, [10]),
    LocationAndSize: createShortcut(ExamplesRecordings.Scenarios.GeoPushpinChart.LocationAndSize),
    LocationAndColor: createShortcut(ExamplesRecordings.Scenarios.GeoPushpinChart.LocationAndColor),
    LocationAndColor_Small: trimmedScenario(
        ExamplesRecordings.Scenarios.GeoPushpinChart.LocationAndColor,
        [-1, 10],
    ),
    LocationSizeAndTooltip: createShortcut(
        ExamplesRecordings.Scenarios.GeoPushpinChart.LocationAndSizeWithTooltip,
    ),
    LocationSizeAndTooltip_Small: trimmedScenario(
        ExamplesRecordings.Scenarios.GeoPushpinChart.LocationAndSizeWithTooltip,
        [-1, 10],
    ),
    LocationSizeAndColor: createShortcut(ExamplesRecordings.Scenarios.GeoPushpinChart.LocationSizeAndColor),
    LocationSizeAndColor_Small: trimmedScenario(
        ExamplesRecordings.Scenarios.GeoPushpinChart.LocationSizeAndColor,
        [-1, 10],
    ),
    LocationSegmentAndColor: createShortcut(
        ExamplesRecordings.Scenarios.GeoPushpinChart.LocationSegmentAndColor,
    ),
    LocationSegmentAndColor_Small: trimmedScenario(
        ExamplesRecordings.Scenarios.GeoPushpinChart.LocationSegmentAndColor,
        [-1, 10],
    ),
    LocationSegmentAndSize: createShortcut(
        ExamplesRecordings.Scenarios.GeoPushpinChart.LocationSegmentAndSize,
    ),
    All: createShortcut(ExamplesRecordings.Scenarios.GeoPushpinChart.LocationSegmentSizeAndColorWithTooltip),
    AllAndSmall: createShortcut(
        ExamplesRecordings.Scenarios.GeoPushpinChart.LocationSegmentSizeAndColorWithTooltipAndFilter,
    ),
};
