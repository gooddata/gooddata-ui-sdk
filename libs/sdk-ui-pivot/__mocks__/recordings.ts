// (C) 2020 GoodData Corporation

import { DataViewFacade } from "@gooddata/sdk-ui";
import { ScenarioRecording, recordedDataView, DataViewAll } from "@gooddata/sdk-backend-mockingbird";

export function recordedDataFacade(rec: ScenarioRecording, dataViewId: string = DataViewAll): DataViewFacade {
    return DataViewFacade.for(recordedDataView(rec, dataViewId));
}
