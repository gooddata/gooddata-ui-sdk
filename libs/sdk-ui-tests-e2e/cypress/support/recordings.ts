// (C) 2022 GoodData Corporation
import { getMockServer } from "./constants";

let testInfo: any;

Cypress.on("test:before:run", (_attributes, test) => {
    testInfo = test;
});

beforeEach(() => {
    const testName = `${testInfo.parent.title} - It ${testInfo.title}`;
    let heading = "TEST:";
    if (testInfo._currentRetry !== 0) {
        heading = "  retry:";
    }
    cy.task("log", `  ${heading} ${testName}`);

    /**
     * Reset scenario state of the mock server
     */
    const mockServer = getMockServer();
    if (mockServer) {
        cy.task("resetRecordingsScenarios", mockServer);
    }

    /**
     * Detect tests that need separate recordings.
     * Separate recordings are needed just for POST and PUT requests.
     * Requests are made unique by adding extra header with test name.
     */
    const separateRecordings = testName.indexOf("(SEPARATE)") !== -1;
    if (separateRecordings) {
        const separateRecordingsKey = testName.replace(/\s/g, "-");
        cy.intercept("POST", "*", (request) => {
            if (request.url.indexOf("executeAfm") === -1) {
                request.headers["X-GDC-TEST-NAME"] = separateRecordingsKey;
            }
        });
        cy.intercept("PUT", "*", (request) => {
            request.headers["X-GDC-TEST-NAME"] = separateRecordingsKey;
        });
    }
});
