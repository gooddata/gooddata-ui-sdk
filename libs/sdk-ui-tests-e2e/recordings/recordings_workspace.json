{
    "workspaceId": "c76e0537d0614abb0027f7c992656b964922506f"
}
