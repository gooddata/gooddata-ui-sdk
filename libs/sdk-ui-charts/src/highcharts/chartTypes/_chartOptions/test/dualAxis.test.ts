// (C) 2019-2020 GoodData Corporation
import { setMeasuresToSecondaryAxis } from "../dualAxis.js";
import { IChartConfig } from "../../../../interfaces/index.js";
import { ReferenceRecordings } from "@gooddata/reference-workspace";
import { recordedDataFacade } from "../../../../../__mocks__/recordings.js";
import { describe, it, expect } from "vitest";

describe("setMeasuresToSecondaryAxis", () => {
    const ComboChart = recordedDataFacade(
        ReferenceRecordings.Scenarios.ComboChart.OnePrimaryAndSecondaryMeasureWithViewBySortedByAttr,
    );
    const dualAxisConfig = { secondary_yaxis: { rotation: "90" } };

    const TEST_CASES = [
        ["", true],
        ["", undefined],
    ];

    it.each(TEST_CASES)(
        "should %s add measures to secondary axis when dualAxis=%s",
        (_desc: string, value: boolean) => {
            const config: IChartConfig = {
                type: "combo2",
                dualAxis: value,
                ...dualAxisConfig,
            };
            expect(setMeasuresToSecondaryAxis(config, ComboChart)).toMatchSnapshot();
        },
    );
});
