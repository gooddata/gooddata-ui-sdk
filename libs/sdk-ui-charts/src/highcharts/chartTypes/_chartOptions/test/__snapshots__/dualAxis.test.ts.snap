// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`setMeasuresToSecondaryAxis > should  add measures to secondary axis when dualAxis=true 1`] = `
{
  "dualAxis": true,
  "secondary_yaxis": {
    "measures": [
      "m_e519fa2a_86c3_4e32_8313_0c03062348j3",
    ],
    "rotation": "90",
  },
  "type": "combo2",
}
`;

exports[`setMeasuresToSecondaryAxis > should  add measures to secondary axis when dualAxis=undefined 1`] = `
{
  "dualAxis": undefined,
  "secondary_yaxis": {
    "measures": [
      "m_e519fa2a_86c3_4e32_8313_0c03062348j3",
    ],
    "rotation": "90",
  },
  "type": "combo2",
}
`;
